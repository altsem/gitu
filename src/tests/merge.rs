use super::*;

fn setup() -> TestContext {
    let ctx = TestContext::setup_clone();
    run(ctx.dir.path(), &["git", "checkout", "-b", "other-branch"]);
    commit(ctx.dir.path(), "new-file", "hello");
    run(ctx.dir.path(), &["git", "checkout", "main"]);
    ctx
}

#[test]
fn merge_menu() {
    snapshot!(setup(), "m");
}

#[test]
fn merge_prompt() {
    snapshot!(setup(), "mm");
}

#[test]
fn merge_ff_only() {
    snapshot!(setup(), "m-fmother-branch<enter>");
}
