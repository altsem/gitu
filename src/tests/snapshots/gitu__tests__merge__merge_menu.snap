---
source: src/tests/merge.rs
expression: ctx.redact_buffer()
---
▌On branch main                                                                 |
▌Your branch is up to date with 'origin/main'.                                  |
                                                                                |
 Recent commits                                                                 |
 b66a0bf main origin/main add initial-file                                      |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
                                                                                |
────────────────────────────────────────────────────────────────────────────────|
Merge                   Arguments                                               |
m merge                 -f Fast-forward only (--ff-only)                        |
a abort                 -n No fast-forward (--no-ff)                            |
c continue                                                                      |
q/<esc> Quit/Close                                                              |
styles_hash: 218a5930a2320238
